src/routes/settings/LightningAddress.tsx

```diff
@@ -0,0 +1,278 @@
+import { Capacitor } from "@capacitor/core";
+import {
+    createForm,
+    required,
+    reset,
+    SubmitHandler
+} from "@modular-forms/solid";
+import { useNavigate } from "@solidjs/router";
+import { Users } from "lucide-solid";
+import {
+    createMemo,
+    createResource,
+    createSignal,
+    Match,
+    Show,
+    Suspense,
+    Switch
+} from "solid-js";
+
+import {
+    BackPop,
+    Button,
+    ButtonCard,
+    DefaultMain,
+    InfoBox,
+    LargeHeader,
+    LightningAddressShower,
+    MutinyPlusCta,
+    MutinyWalletGuard,
+    NavBar,
+    NiceP,
+    TextField,
+    VStack
+} from "~/components";
+import { useI18n } from "~/i18n/context";
+import { useMegaStore } from "~/state/megaStore";
+import { eify } from "~/utils";
+
+type HermesForm = {
+    name: string;
+};
+
+// todo(paul) put this somewhere else
+function HermesForm(props: { onSubmit: (name: string) => void }) {
+    const [state, _] = useMegaStore();
+    const [error, setError] = createSignal<Error>();
+    const [success, setSuccess] = createSignal("");
+
+    const [nameForm, { Form, Field }] = createForm<HermesForm>({
+        initialValues: {
+            name: ""
+        }
+    });
+
+    const network = state.mutiny_wallet?.get_network() || "signet";
+
+    const handleSubmit: SubmitHandler<HermesForm> = async (f: HermesForm) => {
+        setSuccess("");
+        setError(undefined);
+        try {
+            const name = f.name.trim();
+            const available =
+                await state.mutiny_wallet?.check_available_lnurl_name(name);
+            if (!available) {
+                throw new Error("Name already taken");
+            }
+            await state.mutiny_wallet?.reserve_lnurl_name(name);
+            console.log("lnurl name reserved:", name);
+
+            const suffix =
+                network === "signet" ? "@signet.mutiny.plus" : "@mutiny.plus";
+            const formattedName = `${name}${suffix}`;
+
+            const existingProfile = state.mutiny_wallet?.get_nostr_profile();
+
+            const _ = await state.mutiny_wallet?.edit_nostr_profile(
+                undefined,
+                undefined,
+                // lnurl
+                formattedName,
+                // nip05 if they don't have one
+                existingProfile?.nip05 ? undefined : formattedName
+            );
+            reset(nameForm);
+            props.onSubmit(name);
+        } catch (e) {
+            console.error("Error reserving name:", e);
+            setError(eify(e));
+        }
+    };
+
+    return (
+        <Form onSubmit={handleSubmit}>
+            <VStack>
+                <Field name="name" validate={[required("Must not be empty")]}>
```

Comment:
could also validate with the check_available function

Reply:
we also validate it this way on the server, could do the same client side to avoid a round trip

https://github.com/MutinyWallet/blinded-hermes/blob/master/src/register.rs#L17

Reply:
I'm confused I am checking that's available?